include/micm/util/cuda_matrix.cuh

```diff
@@ -9,29 +16,43 @@ namespace micm
     /// @param vectorMatrix Reference to struct containing information about allocated memory
     /// @param num_elements Requested number of elements to allocate
     /// @returns Error code from allocating data on the device, if any
-    int MallocVector(CudaMatrixParam& vectorMatrix, std::size_t num_elements);
+    cudaError_t MallocVector(CudaMatrixParam& vectorMatrix, std::size_t num_elements);
 
     /// @brief Free memory allocated on device
     /// @param vectorMatrix Struct containing allocated device memory
     /// @returns Error code from free-ing data on device, if any
-    int FreeVector(CudaMatrixParam& vectorMatrix);
+    cudaError_t FreeVector(CudaMatrixParam& vectorMatrix);
 
     /// @brief Copies data from the host to the device
     /// @param vectorMatrix Struct containing allocated device memory
     /// @param h_data Host data to copy from
     /// @returns Error code from copying to device from the host, if any
-    int CopyToDevice(CudaMatrixParam& vectorMatrix, std::vector<double>& h_data);
+    cudaError_t CopyToDevice(CudaMatrixParam& vectorMatrix, std::vector<double>& h_data);
 
     /// @brief Copies data from the device to the host
     /// @param vectorMatrix Struct containing allocated device memory
     /// @param h_data Host data to copy data to
     /// @returns Error code from copying from the device to the host, if any
-    int CopyToHost(CudaMatrixParam& vectorMatrix, std::vector<double>& h_data);
+    cudaError_t CopyToHost(CudaMatrixParam& vectorMatrix, std::vector<double>& h_data);
 
     /// @brief Copies data to the destination device memory block from the source device memory block
     /// @param vectorMatrixDest Struct containing allocated destination device memory to copy to
     /// @param vectorMatrixSrc Struct containing allocated source device memory to copy from
     /// @returns Error code from copying to destination device memory from source device memory, if any
-    int CopyToDeviceFromDevice(CudaMatrixParam& vectorMatrixDest, const CudaMatrixParam& vectorMatrixSrc);
+    cudaError_t CopyToDeviceFromDevice(CudaMatrixParam& vectorMatrixDest, const CudaMatrixParam& vectorMatrixSrc);
+
+    /// @brief Checks for CUDA errors and prints error message if any
+    /// @param err Error code to check
+    /// @param file File where error occurred
+    /// @param line Line number where error occurred
+    /// @param str Additional string to print with error message
+    void CHECK_CUDA_ERROR(cudaError_t err, const char *file, int line, std::string str);
```

Comment:
We've been using CamelCase for [function names](https://google.github.io/styleguide/cppguide.html#Function_Names) and reserving ALL_CAPS for [macros](https://google.github.io/styleguide/cppguide.html#Macro_Names) (used sparingly).

This might be a case where a macro could be useful, because you could declare this function as:
```
void CheckCudaError(cudaError_t err, const char *file, int line, std::string str);
```
and then define the macro:
```
#define CUDA_CHECK_ERROR(err, msg) CheckCudaError(err, __FILE__, __LINE__, msg)
```
and then use it as:
```
CHECK_CUDA_ERROR(SomeCudaOp(), "Something went wrong...");
```
and avoid having to use `__FILE__` and `__LINE__` everywhere

Reply:
Thanks for the suggestion. Revised as suggested and hopefully I am doing what you are looking for.